src/index.ts

```diff
@@ -53,12 +53,18 @@ const sidecarPlugin: JupyterFrontEndPlugin<void> = {
             });
           } else {
             const anchor = this.model.get('anchor') || 'right';
+            const ref = this.model.get('ref');
+            const widget_id = ref === null ? null : ref.get('_widget_id');
+            if (ref === null && widget_id !== null) {
+              await ref.created();
+            }
```

Comment:
I'm surprised you're doing `ref.get('_widget_id')` before `await ref.created()`. It looks to me this would be a source of race conditions (widget_id would not be set yet because the ref has not been created yet). I would expect something like this instead?

```suggestion
            const ref = this.model.get('ref');
            let widget_id: string | null = null;
            if (ref) {
              await ref.created();
              widget_id = ref.get('_widget_id');
            }
```

Reply:
Ah sorry! The issue is `ref.created()` -> `ref.created`

`created` is a getter, you need to use it as `ref.created` not call it like a method

Reply:
So indeed 

```typescript
let widget_id: string | null = null;
if (ref) {
  await ref.created;
  widget_id = ref.get('_widget_id');
}
```

Should be the way.

I personally don't think we need `if (ref instanceof SidecarModel)` since we already have this verification Python side